src/structures/paging/mapper/mod.rs

```diff
@@ -82,13 +82,99 @@ pub trait Mapper<S: PageSize> {
     ///
     /// This function might need additional physical frames to create new page tables. These
     /// frames are allocated from the `allocator` argument. At most three frames are required.
+    ///
+    /// Parent page table entries are automatically updated with `PRESENT | WRITABLE | USER_ACCESSIBLE`
+    /// if present in the `PageTableFlags`.
+    ///
+    /// The `map_to_with_table_flags` method gives explicit control over the parent page table flags.
+    ///
+    /// # Examples
+    ///
+    /// Create USER_ACCESSIBLE mapping and update the top hierarchy to be USER_ACCESSIBLE, too:
+    ///
+    /// ```
+    /// # use x86_64::structures::paging::{
+    /// #    Mapper, UnusedPhysFrame, Page, FrameAllocator,
+    /// #    Size4KiB, OffsetPageTable, page_table::PageTableFlags
+    /// # };
+    /// # fn test(mapper: &mut OffsetPageTable, frame_allocator: &mut impl FrameAllocator<Size4KiB>,
+    /// #         page: Page<Size4KiB>, frame: UnusedPhysFrame) {
+    ///         mapper
+    ///           .map_to(
+    ///               page,
+    ///               frame,
+    ///              PageTableFlags::PRESENT
+    ///                   | PageTableFlags::WRITABLE
+    ///                   | PageTableFlags::USER_ACCESSIBLE,
+    ///               frame_allocator,
+    ///           )
+    ///           .unwrap()
+    ///           .flush();
+    /// # }
+    /// ```
+    #[inline]
     fn map_to<A>(
         &mut self,
         page: Page<S>,
         frame: UnusedPhysFrame<S>,
         flags: PageTableFlags,
         frame_allocator: &mut A,
     ) -> Result<MapperFlush<S>, MapToError<S>>
+    where
+        Self: Sized,
+        A: FrameAllocator<Size4KiB>,
+    {
+        let parent_table_flags = flags
+            & (PageTableFlags::PRESENT
+                | PageTableFlags::WRITABLE
+                | PageTableFlags::USER_ACCESSIBLE);
+
+        self.map_to_with_table_flags(page, frame, flags, parent_table_flags, frame_allocator)
+    }
+
+    /// Creates a new mapping in the page table.
+    ///
+    /// This function might need additional physical frames to create new page tables. These
+    /// frames are allocated from the `allocator` argument. At most three frames are required.
+    ///
+    /// The flags of the parent table(s) can be explicitly specified. Those flags are used for
+    /// newly created table entries, and for existing entries the flags are added.
```

Comment:
As above, we should not here that, depending on the used mapper implementation, the PRESENT and WRITABLE flags might be set for parent tables even if they are not specified in `parent_table_flags`.